Civi/Core/LocalizationInitializer.php

```diff
@@ -0,0 +1,160 @@
+ */
+
+namespace Civi\Core;
+
+use Civi;
+use Civi\Core\Event\SystemInstallEvent;
+
+/**
+ * Class LocalizationInitializer
+ * @package Civi\Core
+ */
+class LocalizationInitializer {
+
+  /**
+   * Load the locale settings based on the installation language
+   *
+   * @param \Civi\Core\Event\SystemInstallEvent $event
+   * @throws \CRM_Core_Exception
+   */
+  public static function initialize(SystemInstallEvent $event) {
+
+    // get the current installation language
+    global $tsLocale;
+    $seedLanguage = $tsLocale;
+    if (!$seedLanguage) {
+      return;
+    }
+
+    // get the corresponding settings file if any
+    $localeDir = \CRM_Core_I18n::getResourceDir();
+    $fileName = $localeDir . $seedLanguage . DIRECTORY_SEPARATOR . 'settings.default.json';
+
+    // initalization
+    $settingsParams = array();
+
+    if (file_exists($fileName)) {
+
+      // load the file and parse it
+      $json = file_get_contents($fileName);
+      $settings = json_decode($json, TRUE);
+
+      if (!empty($settings)) {
+        // get all valid settings
+        $results = civicrm_api3('Setting', 'getfields', array());
+        $validSettings = array_keys($results['values']);
+        // add valid settings to params to send to api
+        foreach ($settings as $setting => $value) {
+          if (in_array($setting, $validSettings)) {
+            $settingsParams[$setting] = $value;
+          }
+
+        }
+
+        // ensure we don't mess with multilingual
+        unset($settingsParams['languageLimit']);
+
+        // support for enabled languages (option group)
+        if (isset($settings['languagesOption']) && count($settings['languagesOption']) > 0) {
+          self::updateLanguages($settings['languagesOption']);
+        }
+
+        // set default currency in currencies_enabled (option group)
+        if (isset($settings['defaultCurrency'])) {
+          self::updateCurrencies(array($settings['defaultCurrency']), $settings['defaultCurrency']);
+        }
+
+      }
+
+    }
+
+    // in any case, enforce the seedLanguage as the default language
+    $settingsParams['lcMessages'] = $seedLanguage;
+
+    // apply the config
+    civicrm_api3('Setting', 'create', $settingsParams);
+
+  }
+
+  /**
+   * Enable provided languages and disable all others
+   *
+   * @param $languages array of languages (locale code) ['fr_CA', 'en_CA']
+   * @throws \CiviCRM_API3_Exception
+   */
+  protected static function updateLanguages($languages) {
+    $result = civicrm_api3('OptionValue', 'get', array(
+      'option_group_id' => "languages",
+      'options' => array('limit' => 0),
+    ));
+
+    foreach ($result['values'] as $id => $language) {
+      $params = $language;
+      if (in_array($language['name'], $languages)) {
+        $params['is_active'] = 1;
+      }
+      else {
+        // disable every language that is not in the given list
+        $params['is_active'] = 0;
+      }
+      civicrm_api3('OptionValue', 'create', $params);
```

Comment:
I don't know what others may think ... but I believe that we should care about installation speed because every time someone add some heavy stuff  like this we will end up with installation take very long time... also put into your account that some people may install civicrm on shared servers with limited plans so this would affect them ... ( but maybe I am wrong anyway :) )